src/tools/rest-frontend/README.md

```diff
@@ -0,0 +1,70 @@
+
+The `public` directory contains HTML template files, assets like fonts, compiled JS and CSS files, as well as translation files and all dependencies resolved by `bower`.
+
+## Compiling and Installing
+
+### Dependencies
+
+The project has quite a few dependencies, of which most can be resolved automatically by the used package managers. The only dependency that has to be installed beforehand is the package manager [npm](https://www.npmjs.com/), which comes bundled with [Node.js](https://nodejs.org/) (preferred installation).
+
+### Compiling
+
+The `rest-frontend` has full CMake integration, but does actually only two things:
+- Install (copy) the project files to a target directory.
+- Run `npm install` in this target directory, which does
+ - resolve all `npm` dependencies (into the directory `node_modules`).
+ - resolve all `bower` dependencies (into the directory `public/vendor/bower_components`).
+ - run `grunt full` to compile all application sources (`resources` dir) into working production files (`public` dir).
+
+### Installing
+
+It is not necessary to install anything by hand, CMake does this job already. If changes are made to the source files in `resources`, it is sufficient to run `grunt full` to build the application again. During development, it can be handy to use `grunt watch` to run a watcher daemon that re-compiles LESS or JS files whenever a change was made in the respective `resource` directory.
+
+## Run and Configure
+
+The application allows for some basic configuration. Configurable files can be found in `resources/assets/js/config`, although the only file that should require a change under normal circumstances is the `http.config.js`. It contains the URL to the backend, as well as some URLs for GitHub resources. Any change of this configuration does require to re-run `grunt full` in order to re-compile the project.
+
+To run the application, basically two options are available:
+- Use the built-in webserver of `grunt`, which can be configured in the `Gruntfile.js` and run by `grunt server` (in the installation target directory).
+- Use an own webserver to distribute the application. In order to do so, first `grunt full` should be run. After that, the content of the `public` directory can be copied to any location that suits the needs. `npm` dependencies in the `node_modules` directory and the `resource` directory are only necessary for development, but can be ignored for deployment.
+
+### APIs (Backend & GitHub)
+
+The configuration for API URLs can be found in `resources/assets/js/config/http.config.js`.
+
+### Translations
+
+The configuration for available translations can be found in `resources/assets/js/config/translations.config.js`. To add a translation, copy an existing translation file in `public/assets/translations`, translate it and add the name of the new language to the `translations.config.js` as explained above. After that run `grunt full` to re-compile the application.
```

Comment:
You are right, it does not apply to every situation. But in situations like here, where you give an index where the files are, which content they have or "how to do translations" its practical if you can see the content of the files by following the link.